### ISSUE TITLE
metrics when using custom loss with sparse target categories

### ISSUE THREAD

**Issue by grofte** [keras-team/keras#13552]
**System information**  
- Have I written custom code (as opposed to using example directory):  **YES**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04 
- TensorFlow backend (yes / no):  yes
- TensorFlow version:  '2.0.0'
- Keras version:  '2.3.1'
- Python version:  '3.7.4'
- CUDA/cuDNN version:  
- GPU model and memory: NA 

**Describe the current behavior**  

The `'accuracy'` metric will automatically detect whether you are using a sparse loss function - but only if it is one of the Keras built-in sparse loss functions. And it doesn't print any messages that it is changing behavior. 

**Describe the expected behavior**  

I would either expect that a) `'accuracy'` has one behavior and I would have to change to `'sparse_categorical_accuracy'` when I specify `'sparse_categorical_crossentropy'` or b) print something to standard out every time someone use `'accuracy'` with a sparse loss function. 

Option b makes sense to me. 

Run the code below and flip the comment in "Compile model" to see the two different behaviors. 

**Code to reproduce the issue**  
Provide a reproducible test case that is the bare minimum necessary to generate the problem. 

```python
# multi-class classification with Keras
# https://machinelearningmastery.com/multi-class-classification-tutorial-keras-deep-learning-library/
import pandas
from keras.models import Sequential
from keras.layers import Dense
from keras.wrappers.scikit_learn import KerasClassifier
from keras.utils import np_utils
from sklearn.model_selection import cross_val_score
from sklearn.model_selection import KFold
from sklearn.preprocessing import LabelEncoder
from sklearn.pipeline import Pipeline
import tensorflow as tf
# load dataset
iris_url = "http://archive.ics.uci.edu/ml/machine-learning-databases/iris/iris.data"
dataframe = pandas.read_csv(iris_url, header=None)
dataset = dataframe.values
X = dataset[:,0:4].astype(float)
Y = dataset[:,4]
# encode class values as integers
encoder = LabelEncoder()
encoder.fit(Y)
encoded_Y = encoder.transform(Y)
# convert integers to dummy variables (i.e. one hot encoded)
# dummy_y = np_utils.to_categorical(encoded_Y)

def losses(y_obs, y_pred):
    return tf.keras.losses.sparse_categorical_crossentropy(y_obs, y_pred)

# define baseline model
def baseline_model():
    # create model
    model = Sequential()
    model.add(Dense(8, input_dim=4, activation='relu'))
    model.add(Dense(3, activation='softmax'))
    # Compile model
#     model.compile(loss='sparse_categorical_crossentropy', optimizer='adam', 
#                   metrics=['accuracy', 'sparse_categorical_accuracy'])
    model.compile(loss=losses, optimizer='adam',
                  metrics=['accuracy', 'sparse_categorical_accuracy'])
    return model

estimator = KerasClassifier(build_fn=baseline_model, epochs=200, batch_size=5, verbose=1)
kfold = KFold(n_splits=10, shuffle=True)
results = cross_val_score(estimator, X, encoded_Y, cv=kfold)
print("Baseline: %.2f%% (%.2f%%)" % (results.mean()*100, results.std()*100))
```
